src/core/exec-invoke.c

```diff
@@ -5441,6 +5534,15 @@ int exec_invoke(
         if (r < 0)
                 return r;
 
+        if (context->private_bpf != PRIVATE_BPF_NO) {
+                r = bpffs_finalize(bpffs_socket_fd, &bpffs_pid);
+                if (r < 0) {
+                        *exit_status = EXIT_BPF;
+                        return log_error_errno(r, "Failed to mount bpffs: %m");
+                }
+                TAKE_PIDREF(bpffs_pid);
```

Comment:
TAKE_PIDREF() should be moved to after pidref_wait_for_terminate_and_check() in bpffs_finalize().